src/player/Player.php

```diff
@@ -2255,15 +2256,16 @@ protected function onDeath() : void{
 				$this->getWorld()->dropItem($this->location, $item);
 			}
 
+			$keepOnDeathItemsForInventory = fn(Inventory $inventory) => $inventory->setContents(array_filter($inventory->getContents(), fn(Item $item) => $item->keepOnDeath()));
```

Comment:
This variable name is a bit unpleasant. `$clearInventory` or something similar would be sufficient.